packages/vue-app/template/App.js

```diff
@@ -98,7 +98,12 @@ export default {
   },
   created () {
     // Add this.$nuxt in child instances
-    Vue.prototype.<%= globals.nuxt %> = this
+    const vm = this
+    Vue.mixin({
```

Comment:
Shall we move this mixin to a global scope with a getter proxy? (similar to plugins)

Reply:
Good idea - so only evaluated on access. This is also still not fully working as we need to provide for layout components as well.

Reply:
I've tried various combinations of defining this on the Vue prototype (even the plugin method) but they all reproduce the original bug.

Reply:
@danielroe I've an idea about this, can I update your pr and let you review ? Feel free to revert if it's not proper.

Reply:
Yes, do! @clarkdo

Reply:
@danielroe @pi0 Done, can you review ?